### DISCUSSION TITLE
Script to be used with -watch-with-interval

### DISCUSSION THREAD

**Discussion by scifi1** [icloud-photos-downloader/icloud_photos_downloader#593] [Ideas]
Hi!

Thank you for an awesome solution. This downloader is really useful !

I have a small enhancement request.
Would it be possible to implement a possibility to run a custom script after each completed download run when using the flag
-watch-with-interval ?
I am logging all output to a logfile, and running such a script would be helpful for log rotation and counting number of new files etc

**Comment by AndreyNikiforov** [comment]
> I am logging all output to a logfile, and running such a script would be helpful for log rotation and counting number of new files etc

Curious about your process: what do you use those collected logs for? Do you have backup in addition to log collection?

Alternative to running script from `icloudpd`, you can orchestrate your workflow outside of `icloudpd`, and downloader becomes just one step.

**Reply by scifi1** [reply]
My thought was to look at the logs after each download session to check that everything looks ok.
However, I can also use a post logrotate script so no worries.

Can you recommend an easy to use orchestration framework?

Thanks!

**Reply by AndreyNikiforov** [reply]
> My thought was to look at the logs after each download session to check that everything looks ok. 
> 

So `icloudpd` would run, say, every hour, produce logs (non empty even if nothing changed), and then you'll review those logs periodically (e.g. weekly). I think it would be worth trying that process without automation. Seems to me that there is a lot of human interaction (e.g. open each file and scan it for changes).

I was considering the review process for my backup history. Backup tool/system may be able to show the list of changes for the date interval (e.g. week). My collection is mostly growing and history review is important for catching errors (e.g. if I deleted photo by mistake), so backup history should be short for deleted photos. I haven't done that yet...

> Can you recommend an easy to use orchestration framework?
> 

I haven't looked into that direction.

**Reply by scifi1** [reply]
Thank you. 
Keep on the good work !
